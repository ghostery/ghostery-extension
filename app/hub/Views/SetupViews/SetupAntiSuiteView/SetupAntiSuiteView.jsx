/**
 * Setup Anti-Suite View Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import PropTypes from 'prop-types';
import ClassNames from 'classnames';
import { ToggleSwitch } from '../../../../shared-components';

/**
 * A Functional React component for rendering the Setup Anti-Suite View
 * @return {JSX} JSX for rendering the Setup Anti-Suite View of the Hub app
 * @memberof HubComponents
 */
const SetupAntiSuiteView = props => (
	<div className="row align-center">
		<div className="columns small-12 medium-10 large-8">
			{props.features.map((feature) => {
				const iconClassNames = ClassNames(feature.id, {
					SetupAntiSuite__icon: true,
					active: feature.enabled,
				});

				return (
					<div key={`feature-${feature.id}`} className="SetupAntiSuite__feature row align-center-middle">
						<div className="columns shrink">
							<div className={iconClassNames} />
						</div>
						<div className="columns shrink">
							<ToggleSwitch
								checked={feature.enabled}
								onChange={feature.toggle}
							/>
						</div>
						<div className="columns small-12 large-8">
							<div className="flex-container align-middle">
								<div className="SetupAntiSuite__feature_title display-inline">
									{feature.name}
								</div>
								{feature.enabled && (
									<div className="SetupAntiSuite__feature_enabled display-inline">
										{t('hub_setup_antisuite_feature_enabled')}
									</div>
								)}
							</div>
							<div className="SetupAntiSuite__feature_description">
								{feature.description}
							</div>
						</div>
					</div>
				);
			})}
		</div>
	</div>
);

// PropTypes ensure we pass required props of the correct type
SetupAntiSuiteView.propTypes = {
	features: PropTypes.arrayOf(PropTypes.shape({
		id: PropTypes.string.isRequired,
		name: PropTypes.string.isRequired,
		enabled: PropTypes.bool.isRequired,
		toggle: PropTypes.func.isRequired,
		description: PropTypes.string.isRequired,
	})).isRequired,
};

export default SetupAntiSuiteView;
