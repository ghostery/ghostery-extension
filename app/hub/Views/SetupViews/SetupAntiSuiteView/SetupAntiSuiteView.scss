/**
 * Setup Anti-Suite View Sass
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

// Setup Anti-Suite View
.SetupAntiSuite__feature {
	padding: 30px 0 20px;
	border-bottom: 1px solid #cbd4d7;
}
.SetupAntiSuite__feature:last-of-type {
	border-bottom: none;
}
.SetupAntiSuite__feature_title {
	font-size: 17.3px;
	font-weight: 500;
	color: #4a4a4a;
}
.SetupAntiSuite__feature_enabled {
	color: #1dafed;
	font-size: 12.5px;
	font-weight: 500;
	margin-left: 50px;
}
.SetupAntiSuite__feature_description {
	font-size: 10.6px;
	line-height: 1.75;
	margin: 15px 0 10px;
	color: #4a4a4a;
}
.SetupAntiSuite__icon {
	height: 50px;
	width: 50px;
	margin: 0 auto;
	background-repeat: no-repeat;
	background-position: center center;
}

// Anti-Tracking Icon
.SetupAntiSuite__feature .anti-tracking {
	background-image: buildAntiTrackingIcon(#9f9f9f);
}
.SetupAntiSuite__feature .anti-tracking.active {
	background-image: buildAntiTrackingActiveIcon(#1dafed);
}

// Ad Block Icon
.SetupAntiSuite__feature .ad-block {
	background-image: buildAdBlockIcon(#9f9f9f);
}
.SetupAntiSuite__feature .ad-block.active {
	background-image: buildAdBlockActiveIcon(#1dafed);
}

// Smart Blocking Icon
.SetupAntiSuite__feature .smart-blocking {
	background-image: buildSmartBlockingIcon(#9f9f9f);
}
.SetupAntiSuite__feature .smart-blocking.active {
	background-image: buildSmartBlockingActiveIcon(#1dafed);
}

// Ghostery Rewards Icon
.SetupAntiSuite__feature .ghostery-rewards {
	background-image: buildGhosteryRewardsIcon(#9f9f9f);
}
.SetupAntiSuite__feature .ghostery-rewards.active {
	background-image: buildGhosteryRewardsActiveIcon(#1dafed);
}
