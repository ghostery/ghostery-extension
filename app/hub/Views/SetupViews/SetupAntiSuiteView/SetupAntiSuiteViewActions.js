/**
 * Setup Anti-Suite View Action creators
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2019 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import { log, sendMessageInPromise } from '../../../utils';
import {
	SET_ANTI_TRACKING,
	SET_AD_BLOCK,
	SET_SMART_BLOCK
} from '../../SetupView/SetupViewConstants';

export function setAntiTracking(actionData) {
	return function(dispatch) {
		return sendMessageInPromise(SET_ANTI_TRACKING, actionData).then((data) => {
			dispatch({
				type: SET_ANTI_TRACKING,
				data,
			});
		}).catch((err) => {
			log('setupBlocking Action setAntiTracking Error', err);
		});
	};
}

export function setAdBlock(actionData) {
	return function(dispatch) {
		return sendMessageInPromise(SET_AD_BLOCK, actionData).then((data) => {
			dispatch({
				type: SET_AD_BLOCK,
				data,
			});
		}).catch((err) => {
			log('setupBlocking Action setAdBlock Error', err);
		});
	};
}

export function setSmartBlocking(actionData) {
	return function(dispatch) {
		return sendMessageInPromise(SET_SMART_BLOCK, actionData).then((data) => {
			dispatch({
				type: SET_SMART_BLOCK,
				data,
			});
		}).catch((err) => {
			log('setupBlocking Action setSmartBlocking Error', err);
		});
	};
}
