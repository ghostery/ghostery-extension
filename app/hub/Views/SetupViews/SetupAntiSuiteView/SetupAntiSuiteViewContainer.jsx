/**
 * Setup Anti-Suite View Container
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import SetupAntiSuiteView from './SetupAntiSuiteView';

/**
 * @class Implement the Setup Anti-Suite View for the Ghostery Hub
 * @extends Component
 * @memberof HubComponents
 */
class SetupAntiSuiteViewContainer extends Component {
	/**
	 * Lifecycle Event
	 */
	componentWillMount() {
		const title = t('hub_setup_page_title_antisuite');
		window.document.title = title;

		const { index, setup, sendMountActions } = this.props;
		this.props.actions.setSetupNavigation({
			activeIndex: index,
			hrefPrev: `/setup/${index - 1}`,
			hrefNext: `/setup/${index + 1}`,
			hrefDone: '/',
			textPrev: t('hub_setup_nav_previous'),
			textNext: t('hub_setup_nav_next'),
			textDone: t('hub_setup_exit_flow'),
		});

		if (sendMountActions) {
			const {
				enable_anti_tracking,
				enable_ad_block,
				enable_smart_blocking,
				enable_ghostery_rewards,
			} = setup;
			this.props.actions.setAntiTracking({ enable_anti_tracking });
			this.props.actions.setAdBlock({ enable_ad_block });
			this.props.actions.setSmartBlocking({ enable_smart_blocking });
			this.props.actions.setGhosteryRewards({ enable_ghostery_rewards });
		}
	}

	/**
	* Function to handle toggling a feature on the Setup Anti-Suite View
	* @param  {Object} featureName the name of the feature being toggled
	*/
	_handleToggle = (featureName) => {
		switch (featureName) {
			case 'anti-tracking': {
				const enable_anti_tracking = !this.props.setup.enable_anti_tracking;
				this.props.actions.setAntiTracking({ enable_anti_tracking });
				break;
			}
			case 'ad-block': {
				const enable_ad_block = !this.props.setup.enable_ad_block;
				this.props.actions.setAdBlock({ enable_ad_block });
				break;
			}
			case 'smart-blocking': {
				const enable_smart_blocking = !this.props.setup.enable_smart_blocking;
				this.props.actions.setSmartBlocking({ enable_smart_blocking });
				break;
			}
			case 'ghostery-rewards': {
				const enable_ghostery_rewards = !this.props.setup.enable_ghostery_rewards;
				this.props.actions.setGhosteryRewards({ enable_ghostery_rewards });
				break;
			}
			default: break;
		}
	}

	/**
	 * React's required render function. Returns JSX
	 * @return {JSX} JSX for rendering the Setup Anti-Suite View of the Hub app
	 */
	render() {
		const {
			enable_anti_tracking,
			enable_ad_block,
			enable_smart_blocking,
			enable_ghostery_rewards,
		} = this.props.setup;
		const features = [
			{
				id: 'anti-tracking',
				name: t('hub_setup_antisuite_name_antitracking'),
				enabled: enable_anti_tracking,
				toggle: () => {
					this._handleToggle('anti-tracking');
				},
				description: t('hub_setup_antisuite_description_antitracking'),
			},
			{
				id: 'ad-block',
				name: t('hub_setup_adblock_name_adblocking'),
				enabled: enable_ad_block,
				toggle: () => {
					this._handleToggle('ad-block');
				},
				description: t('hub_setup_adblock_description_adblocking'),
			},
			{
				id: 'smart-blocking',
				name: t('hub_setup_smartblocking_name_smartblocking'),
				enabled: enable_smart_blocking,
				toggle: () => {
					this._handleToggle('smart-blocking');
				},
				description: t('hub_setup_smartblocking_description_smartblocking'),
			},
			{
				id: 'ghostery-rewards',
				name: t('hub_setup_ghosteryrewards_name_rewards'),
				enabled: enable_ghostery_rewards,
				toggle: () => {
					this._handleToggle('ghostery-rewards');
				},
				description: t('hub_setup_ghosteryrewards_description_rewards'),
			},
		];

		return <SetupAntiSuiteView features={features} />;
	}
}

// PropTypes ensure we pass required props of the correct type
SetupAntiSuiteViewContainer.propTypes = {
	index: PropTypes.number.isRequired,
	actions: PropTypes.shape({
		setSetupNavigation: PropTypes.func.isRequired,
		setAntiTracking: PropTypes.func.isRequired,
		setAdBlock: PropTypes.func.isRequired,
		setSmartBlocking: PropTypes.func.isRequired,
		setGhosteryRewards: PropTypes.func.isRequired,
	}).isRequired,
};

export default SetupAntiSuiteViewContainer;
