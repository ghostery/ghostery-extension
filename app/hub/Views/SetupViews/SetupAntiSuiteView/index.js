/**
 * Point of entry index.js file for Setup Anti-Suite View
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import SetupAntiSuiteViewContainer from './SetupAntiSuiteViewContainer';
import * as SetupAntiSuiteViewActions from './SetupAntiSuiteViewActions';
import { setSetupNavigation } from '../../SetupView/SetupViewActions';

/**
 * Map redux store state properties to the component's own properties.
 * @param  {Object} state    entire Redux store's state
 * @return {function}        this function returns a plain object, which will be merged into the component's props
 * @memberof HubContainers
 */
const mapStateToProps = state => Object.assign({}, state.setup);

/**
 * Bind the component's action creators using Redux's bindActionCreators.
 * @param  {function} dispatch redux store method which dispatches actions
 * @return {function}          to be used as an argument in redux connect call
 * @memberof SetupContainers
 */
const mapDispatchToProps = dispatch => ({
	actions: bindActionCreators(Object.assign(SetupAntiSuiteViewActions, { setSetupNavigation }), dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(SetupAntiSuiteViewContainer);
