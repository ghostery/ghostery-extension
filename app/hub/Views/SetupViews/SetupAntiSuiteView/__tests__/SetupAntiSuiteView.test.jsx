/**
 * Setup Anti-Suite View Test Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import SetupAntiSuiteView from '../SetupAntiSuiteView';

describe('app/hub/Views/SetupViews/SetupAntiSuiteView component', () => {
	describe('Snapshot tests with react-test-renderer', () => {
		test('setup anti-suite view is rendered correctly', () => {
			const initialState = {
				features: [
					{
						id: 'test-feature-1',
						name: 'Test Feature One',
						enabled: true,
						toggle: () => {},
						description: 'The first test feature',
					},
					{
						id: 'test-feature-2',
						name: 'Test Feature Two',
						enabled: false,
						toggle: () => {},
						description: 'The second test feature',
					},
				],
			};

			const component = renderer.create(
				<SetupAntiSuiteView {...initialState} />
			).toJSON();
			expect(component).toMatchSnapshot();
		});
	});

	describe('More Snapshot tests with react-test-renderer, but for edge cases', () => {
		test('edge case where features is an empty array', () => {
			const initialState = {
				features: [],
			};

			const component = renderer.create(
				<SetupAntiSuiteView {...initialState} />
			).toJSON();
			expect(component).toMatchSnapshot();
		});
	});

	describe('Shallow snapshot tests rendered with Enzyme', () => {
		test('the happy path of the component', () => {
			const initialState = {
				features: [
					{
						id: 'test-feature-1',
						name: 'Test Feature One',
						enabled: true,
						toggle: () => {},
						icon: '',
						description: 'The first test feature',
					},
					{
						id: 'test-feature-2',
						name: 'Test Feature Two',
						enabled: false,
						toggle: () => {},
						icon: '',
						description: 'The second test feature',
					},
				],
			};

			const component = shallow(<SetupAntiSuiteView {...initialState} />);
			expect(component.find('.row').length).toBe(3);
			expect(component.find('.test-feature-1').length).toBe(1);
			expect(component.find('.test-feature-2').length).toBe(1);
			expect(component.find('.test-feature-1.active').length).toBe(1);
			expect(component.find('.test-feature-2.active').length).toBe(0);
		});

		test('the non-happy path of the component', () => {
			const initialState = {
				features: [],
			};

			const component = shallow(<SetupAntiSuiteView {...initialState} />);
			expect(component.find('.row').length).toBe(1);
		});
	});
});
