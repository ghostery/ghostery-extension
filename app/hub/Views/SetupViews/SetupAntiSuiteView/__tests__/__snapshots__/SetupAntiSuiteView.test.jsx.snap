// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/hub/Views/SetupViews/SetupAntiSuiteView component More Snapshot tests with react-test-renderer, but for edge cases edge case where features is an empty array 1`] = `
<div
  className="row align-center"
>
  <div
    className="columns small-12 medium-10 large-8"
  />
</div>
`;

exports[`app/hub/Views/SetupViews/SetupAntiSuiteView component Snapshot tests with react-test-renderer setup anti-suite view is rendered correctly 1`] = `
<div
  className="row align-center"
>
  <div
    className="columns small-12 medium-10 large-8"
  >
    <div
      className="SetupAntiSuite__feature row align-center-middle"
    >
      <div
        className="columns shrink"
      >
        <div
          className="test-feature-1 SetupAntiSuite__icon active"
        />
      </div>
      <div
        className="columns shrink"
      >
        <div
          className="ToggleSwitch ToggleSwitch--active"
          onClick={[Function]}
        >
          <div
            className="ToggleSwitch__bar"
          />
          <span
            className="ToggleSwitch__circle"
          />
        </div>
      </div>
      <div
        className="columns small-12 large-8"
      >
        <div
          className="flex-container align-middle"
        >
          <div
            className="SetupAntiSuite__feature_title display-inline"
          >
            Test Feature One
          </div>
          <div
            className="SetupAntiSuite__feature_enabled display-inline"
          >
            hub_setup_antisuite_feature_enabled
          </div>
        </div>
        <div
          className="SetupAntiSuite__feature_description"
        >
          The first test feature
        </div>
      </div>
    </div>
    <div
      className="SetupAntiSuite__feature row align-center-middle"
    >
      <div
        className="columns shrink"
      >
        <div
          className="test-feature-2 SetupAntiSuite__icon"
        />
      </div>
      <div
        className="columns shrink"
      >
        <div
          className="ToggleSwitch"
          onClick={[Function]}
        >
          <div
            className="ToggleSwitch__bar"
          />
          <span
            className="ToggleSwitch__circle"
          />
        </div>
      </div>
      <div
        className="columns small-12 large-8"
      >
        <div
          className="flex-container align-middle"
        >
          <div
            className="SetupAntiSuite__feature_title display-inline"
          >
            Test Feature Two
          </div>
        </div>
        <div
          className="SetupAntiSuite__feature_description"
        >
          The second test feature
        </div>
      </div>
    </div>
  </div>
</div>
`;
